test/e2e/main_test.go

```diff
@@ -3,57 +3,245 @@ package e2e
+						Config: &metering.ReportingOperatorConfig{
+							LogLevel: "debug",
+							Prometheus: &metering.ReportingOperatorPrometheusConfig{
+								MetricsImporter: &metering.ReportingOperatorPrometheusMetricsImporterConfig{
+									Config: &metering.ReportingOperatorPrometheusMetricsImporterConfigSpec{
+										ChunkSize:                 &meta.Duration{Duration: 5 * time.Minute},
+										PollInterval:              &meta.Duration{Duration: 30 * time.Second},
+										StepSize:                  &meta.Duration{Duration: 1 * time.Minute},
+										MaxImportBackfillDuration: &meta.Duration{Duration: 15 * time.Minute},
+										MaxQueryRangeDuration:     &meta.Duration{Duration: 5 * time.Minute},
+									},
+								},
+							},
+						},
+					},
+				},
+				Presto: &metering.Presto{
+					Spec: &metering.PrestoSpec{
+						Coordinator: &metering.PrestoCoordinatorSpec{
+							Resources: &v1.ResourceRequirements{
+								Requests: v1.ResourceList{
+									v1.ResourceCPU:    resource.MustParse("1"),
+									v1.ResourceMemory: resource.MustParse("1Gi"),
+								},
+							},
+						},
+					},
+				},
+				Hive: &metering.Hive{
+					Spec: &metering.HiveSpec{
+						Metastore: &metering.HiveMetastoreSpec{
+							Resources: &v1.ResourceRequirements{
+								Requests: v1.ResourceList{
+									v1.ResourceCPU:    resource.MustParse("1"),
+									v1.ResourceMemory: resource.MustParse("650Mi"),
+								},
+							},
+							Storage: &metering.HiveMetastoreStorageConfig{
+								Size: "5Gi",
+							},
+						},
+						Server: &metering.HiveServerSpec{
+							Resources: &v1.ResourceRequirements{
+								Requests: v1.ResourceList{
+									v1.ResourceCPU:    resource.MustParse("500m"),
+									v1.ResourceMemory: resource.MustParse("650Mi"),
+								},
+							},
+						},
+					},
+				},
+				Hadoop: &metering.Hadoop{
+					Spec: &metering.HadoopSpec{
+						HDFS: &metering.HadoopHDFS{
+							Enabled: testhelpers.PtrToBool(true),
+							Datanode: &metering.HadoopHDFSDatanodeSpec{
+								Resources: &v1.ResourceRequirements{
+									Requests: v1.ResourceList{
+										v1.ResourceMemory: resource.MustParse("500Mi"),
+									},
+								},
+								Storage: &metering.HadoopHDFSStorageConfig{
+									Size: "5Gi",
+								},
+							},
+							Namenode: &metering.HadoopHDFSNamenodeSpec{
+								Resources: &v1.ResourceRequirements{
+									Requests: v1.ResourceList{
+										v1.ResourceMemory: resource.MustParse("500Mi"),
+									},
+								},
+								Storage: &metering.HadoopHDFSStorageConfig{
+									Size: "5Gi",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, testCase := range testInstallConfigs {
+		t := t
+		testCase := testCase
+
+		t.Run(testCase.Name, func(t *testing.T) {
+			testInstall(t, testCase.MeteringOperatorImageRepo, testCase.MeteringOperatorImageTag, testCase.Name, testCase.TargetPods, testCase.MeteringConfigSpec)
+		})
+	}
+}
+
+func testInstall(
```

Comment:
We'll want to figure out a better name for this, but I also think it'll change a bit once it needs to run other sub-tests besides `testReportingProducesData`